src/common/buffer.cc

```diff
@@ -1350,7 +1350,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
 
   // -- buffer::list --
 
-  buffer::list::list(list&& other)
+  buffer::list::list(list&& other) noexcept
     : _buffers(std::move(other._buffers)),
```

Comment:
@stiopaa1 yeah, the default `std::allocator`'s ctor is not `noexcept`, and the list's sentry node could be allocated on heap. 

agree with you that it's safe to make sure that every line of the function should be `noexcept` if the function is `noexcept`. but i think the `noexcept` also defines an "exception boundary".

what if we can/do not handle the `bad_alloc` exception? can we add the `noexcept` specifier to the function which throws `bad_alloc` for better performance?

`buffer::list` is both a building block of ceph, it's also a part of librados library. so it is not acceptable for it to make a decision about killing the program on OOM. so as you suggested, i revert this change. but i think we can have it in other part of our project. for example, in #9361.